cmd/geth/dbcmd.go

```diff
@@ -352,7 +352,7 @@ func inspectTrie(ctx *cli.Context) error {
 	if ctx.NArg() >= 1 {
 		if ctx.Args().Get(0) == "latest" {
 			headerHash := rawdb.ReadHeadHeaderHash(db)
-			blockNumber = *(rawdb.ReadHeaderNumber(db, headerHash))
+			blockNumber = *(rawdb.ReadHeaderNumber(db.BlockStore(), headerHash))
```

Comment:
Need to judge blockStore e! = nil

Reply:
```
func (frdb *freezerdb) BlockStore() ethdb.KeyValueStore {
	if frdb.blockStore != nil {
		return frdb.blockStore
	} else {
		return frdb
	}
}
```

if the blockstore is nil, the frdb will be returned.